robot/impl/resource_manager.go

```diff
@@ -819,50 +819,65 @@ func (manager *resourceManager) completeConfig(
 		if !ok || !gNode.NeedsReconfigure() {
 			continue
 		}
-		var verb string
-		if gNode.IsUninitialized() {
-			verb = "configuring"
-		} else {
-			verb = "reconfiguring"
-		}
-		manager.logger.CInfow(ctx, fmt.Sprintf("Now %s a remote", verb), "resource", resName)
-		switch resName.API {
-		case client.RemoteAPI:
-			remConf, err := resource.NativeConfig[*config.Remote](gNode.Config())
-			if err != nil {
-				manager.logger.CErrorw(ctx, "remote config error", "error", err)
-				continue
-			}
+		processAndCompleteConfigForRemote := func() {
+			var verb string
 			if gNode.IsUninitialized() {
-				gNode.InitializeLogger(
-					manager.logger, fromRemoteNameToRemoteNodeName(remConf.Name).String(),
-				)
-			}
-			// this is done in config validation but partial start rules require us to check again
-			if _, err := remConf.Validate(""); err != nil {
-				gNode.LogAndSetLastError(
-					fmt.Errorf("remote config validation error: %w", err), "remote", remConf.Name)
-				continue
+				verb = "configuring"
+			} else {
+				verb = "reconfiguring"
 			}
-			rr, err := manager.processRemote(ctx, *remConf, gNode)
-			if err != nil {
-				gNode.LogAndSetLastError(
-					fmt.Errorf("error connecting to remote: %w", err), "remote", remConf.Name)
-				continue
+			manager.logger.CInfow(ctx, fmt.Sprintf("Now %s a remote", verb), "resource", resName)
+			switch resName.API {
+			case client.RemoteAPI:
+				remConf, err := resource.NativeConfig[*config.Remote](gNode.Config())
+				if err != nil {
+					manager.logger.CErrorw(ctx, "remote config error", "error", err)
+					return
+				}
+				if gNode.IsUninitialized() {
+					gNode.InitializeLogger(
+						manager.logger, fromRemoteNameToRemoteNodeName(remConf.Name).String(),
+					)
+				}
+				// this is done in config validation but partial start rules require us to check again
+				if _, err := remConf.Validate(""); err != nil {
+					gNode.LogAndSetLastError(
+						fmt.Errorf("remote config validation error: %w", err), "remote", remConf.Name)
+					return
+				}
+				rr, err := manager.processRemote(ctx, *remConf, gNode)
+				if err != nil {
+					gNode.LogAndSetLastError(
+						fmt.Errorf("error connecting to remote: %w", err), "remote", remConf.Name)
+					return
+				}
+				manager.addRemote(ctx, rr, gNode, *remConf)
+				rr.SetParentNotifier(func() {
+					lr.sendTriggerConfig(remConf.Name)
+				})
+			default:
+				err := errors.New("config is not a remote config")
+				manager.logger.CErrorw(ctx, err.Error(), "resource", resName)
 			}
-			manager.addRemote(ctx, rr, gNode, *remConf)
-			rr.SetParentNotifier(func() {
-				lr.sendTriggerConfig(remConf.Name)
-			})
-		default:
-			err := errors.New("config is not a remote config")
-			manager.logger.CErrorw(ctx, err.Error(), "resource", resName)
 		}
+		lr.reconfigureWorkers.Add(1)
+		remoteErrGroup.Go(func() error {
+			defer lr.reconfigureWorkers.Done()
+			processAndCompleteConfigForRemote()
+			return nil
```

Comment:
To be clear: it looks like you're not leveraging the error-returning capacity of any of these `Go`ed goroutines? 

Seems fine to me, as I'm guessing you're just trying to leverage the `SetLimit` API you use upon creation of the `errgroup.Group`. We do have our _own_ type called [`StoppableWorkers`](https://github.com/viamrobotics/goutils/blob/main/stoppable_workers.go#L11) that's come in handy, but it lacks anything like `SetLimit`. That makes me wonder if a) we should add that functionality and b) whether there's some equivalent to `errgroup.Group` that takes `func()` instead of `func() error`. I think the answers are no and no but thought I'd bring it up.